enumerable.rb

```diff
@@ -0,0 +1,112 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    for x in self
+      yield(x)
+    end
+    x
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    x = 0
+    for y in self
+      yield(x, y)
+      x += 1
+    end
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    new_array = []
+    my_each do |x|
+      new_array.push(x) if yield(x)
+    end
+    new_array
+  end
+
+  def my_all?
+    return enum_for unless block_given?
+
+    count = 0
+    my_each do |x|
+      count += 1 if yield(x)
```

Comment:
- [ ] my_all? Should also work without any block or argument passed. EX:
```ruby
[false, true, 1, 'ok'].my_all?
=> false
[{a: 1}, true, 1, 'ok'].my_all?
=> true
```

Reply:
In that case it returns true if all elements of the enumerable are **truthy**

Reply:
- [ ] my_all should also accept just a param, and no block, and check if it matches the pattern. The pattern can be a Class, a Regular Expression, or any thing you can compare with '==' too.